### PR TITLE
Create endicia package

### LINKED ISSUE #1: Create endicia package
Create an endicia package with dependencies on http-service and endicia-service

### PR DESCRIPTION
**Pull request by agreggio-slingr**
Pull-request for issue #15473 created by @agreggio-slingr 

## What it does?

Create endicia package

## How to test it?

You can install the following test [package](https://github.com/slingr-stack/endicia-package-test) with the following credentials.

Client ID: zGNzowNmPCPPTqmKfjSEPplfnW133rwZ
Client Secret: GPX_PTHFBolg5WH5U91sX-PiDsI4QbJDjOit-T2FhcqDmmU2rgDlvACWdhsphcfY
Code Verifier : 7f410047e678b110982e42c1c9489e5a365957407e72552da8e68afe

Username: slingr-01
Password: December2023!
```
var response = pkg.endicia.api.get('/balance')
log(JSON.stringify(response))
```

## Implementation notes

N/A

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,20 +19,20 @@
...
 
-The Javascript API of the skeleton endpoint has two pieces:
+The Javascript API of the endicia endpoint has two pieces:
 
 - **HTTP requests**
 - **Flow steps**
 
 ## HTTP requests
-You can make `GET`,`PUT`,`PATCH`,`DELETE` requests to the [skeleton API](API_URL_HERE) like this:
+You can make `GET`,`PUT`,`PATCH`,`DELETE` requests to the [endicia API](API_URL_HERE) like this:
```
Comment (pasaperez-work) [inline README.md]: Replace the API_URL_HERE by the link to documentation.

File: README.md [hunk 2]
```diff
@@ -19,20 +19,20 @@
...
 ```javascript
-var response = pkg.skeleton.api.get('/path3')
-var response = pkg.skeleton.api.put('/path1/:testPath', body)
-var response = pkg.skeleton.api.put('/path1/:testPath')
-var response = pkg.skeleton.api.patch('/path2?param2=' + httpOptions.query.param2 + '&param3=' + httpOptions.query.param3 + '', body)
-var response = pkg.skeleton.api.patch('/path2?param2=' + httpOptions.query.param2 + '&param3=' + httpOptions.query.param3 + '')
-var response = pkg.skeleton.api.delete('/path4')
+var response = pkg.endicia.api.get('/path3')
```
Comment (pasaperez-work) [inline README.md]: Review this paths, add at least a couple, for example the same path witch you test in this PR.

File: README.md [hunk 3]
```diff
@@ -205,7 +205,6 @@ For more information about how shortcuts or flow steps work, and how they are ge
 
 ## Dependencies
 * HTTP Service (Latest Version)
```
Comment (pasaperez-work) [inline README.md]:
```suggestion
* HTTP Service (v1.3.7 Version)
```

File: flowSteps/apiCallEndicia/step.js [hunk 4]
```diff
@@ -117,86 +142,70 @@ function stringToObject (obj) {
...
-		},
-		config.get("privateKey"),
-		"RS256"
-	)
 }
 
 function mergeJSON (json1, json2) {
-	var result = {};
-	var key;
+	const result = {};
```
Comment (pasaperez-work) [inline flowSteps/apiCallEndicia/step.js]:
```suggestion
	var result = {};
```

File: flowSteps/apiCallEndicia/step.js [hunk 5]
```diff
@@ -117,86 +142,70 @@ function stringToObject (obj) {
...
+	let key;
```
Comment (pasaperez-work) [inline flowSteps/apiCallEndicia/step.js]:
```suggestion
	var key;
```

File: flowSteps/apiCallEndicia/step.js [hunk 6]
```diff
@@ -205,3 +214,20 @@ function mergeJSON (json1, json2) {
 	}
 	return result;
 }
+
+/****************************************************
+ Extra helper
+ ****************************************************/
+
+exports.callbackTest = function () {
```
Comment (pasaperez-work) [inline flowSteps/apiCallEndicia/step.js:222]: You can remove this call

File: package.json [hunk 7]
```diff
@@ -78,20 +93,10 @@
...
-        },
         {
             "type": "flowStep",
-            "namespace": "customFlow",
-            "path": "/flowSteps/customFlow"
+            "namespace": "apiCallEndicia",
+            "path": "/flowSteps/apiCallEndicia"
         },
         {
             "type": "uiService",
```
Comment (pasaperez-work) [inline package.json]: You don't use this uiService

**Review by angel-perez-itti — CHANGES_REQUESTED** [review]

**Comment by pasaperez-work** [comment]
#approve

**Comment by pasaperez-work** [comment]
Retested on Test Packages of tarzan, the extenal system or the credentials could be expired.

**Pull request merged**